components/web_discovery/browser/server_config_loader.cc

```diff
@@ -0,0 +1,446 @@
+      description:
+        "Requests server configuration needed to send Web Discovery "
+        "measurements to Brave servers."
+      trigger:
+        "Requests are automatically sent at intervals "
+        "while Brave is running."
+      data: "Configuration attributes"
+      destination: WEBSITE
+    }
+    policy {
+      cookies_allowed: NO
+      setting:
+        "Users can opt-in or out via brave://settings/search"
+    })");
+
+constexpr char kGroupPubKeysFieldName[] = "groupPubKeys";
+constexpr char kPubKeysFieldName[] = "pubKeys";
+constexpr char kMinVersionFieldName[] = "minVersion";
+constexpr char kKeysFieldName[] = "keys";
+constexpr char kLimitFieldName[] = "limit";
+constexpr char kPeriodFieldName[] = "period";
+constexpr char kSourceMapFieldName[] = "sourceMap";
+constexpr char kSourceMapActionsFieldName[] = "actions";
+constexpr char kLocationFieldName[] = "location";
+
+constexpr char kCollectorConfigPathWithFields[] =
+    "/config?fields=minVersion,groupPubKeys,pubKeys,sourceMap";
+constexpr char kQuorumConfigPath[] = "/config";
+constexpr char kPatternsFilename[] = "wdp_patterns.json";
+
+constexpr char kOmittedLocationValue[] = "--";
+constexpr auto kAllowedReportLocations =
+    base::MakeFixedFlatSet<std::string_view>(
+        {"ar", "at", "au", "be", "br", "ca", "ch", "cn", "cz", "de",
+         "dk", "es", "fi", "fr", "gb", "gr", "hu", "in", "it", "jp",
+         "mx", "nl", "no", "pl", "ro", "ru", "se", "ua", "us"});
+
+KeyMap ParseKeys(const base::Value::Dict& encoded_keys) {
+  KeyMap map;
+  for (const auto [date, key_b64] : encoded_keys) {
+    std::vector<uint8_t> decoded_data;
+    // Decode to check for valid base64
+    if (!base::Base64Decode(key_b64.GetString())) {
+      continue;
+    }
+    map[date] = key_b64.GetString();
+  }
+  return map;
+}
+
+base::flat_map<std::string, std::unique_ptr<SourceMapActionConfig>>
+ParseSourceMapActionConfigs(const base::Value::Dict& configs_dict) {
+  base::flat_map<std::string, std::unique_ptr<SourceMapActionConfig>> map;
+  for (const auto [action, config_dict_val] : configs_dict) {
+    auto* config_dict = config_dict_val.GetIfDict();
+    if (!config_dict) {
+      continue;
+    }
+    auto& action_config = map[action];
+    if (!action_config) {
+      action_config = std::make_unique<SourceMapActionConfig>();
+    }
+    auto* keys_list = config_dict->FindList(kKeysFieldName);
+    if (keys_list) {
+      for (const auto& key_val : *keys_list) {
+        if (key_val.is_string()) {
+          action_config->keys.push_back(key_val.GetString());
+        }
+      }
+    }
+    auto limit = config_dict->FindInt(kLimitFieldName);
+    auto period = config_dict->FindInt(kPeriodFieldName);
+
+    action_config->limit = limit && limit > 0 ? *limit : 1;
+    action_config->period = period && period > 0 ? *period : 24;
+  }
+  return map;
+}
+
+std::optional<std::string> GunzipContents(std::string gzipped_contents) {
```

Comment:
why this machinery is required? Is server missing `Content-Encoding: gzip` header?

`URLLoader` should handle this automatically. There's a test: `SimpleURLLoaderTest, GzipBody`.

Reply:
> Is server missing `Content-Encoding: gzip` header?

Yes, the server does not provide the header since it's serving a pre-gzipped file.

Reply:
> > Is server missing `Content-Encoding: gzip` header?
> 
> Yes, the server does not provide the header because it's serving a pre-gzipped file.

Sorry, but that's not an excuse. Is it served from AWS/Cloudflare? Just set "Content-Encoding: gzip" on server, do not perform manual unzipping.

example how to set during `aws s3 cp`: https://github.com/brave/brave-variations/blob/55500374116ea45c0c1cd9e3cb2416326c80a285/.github/workflows/generate-test-seed.yml#L82

Reply:
that modification may have ramifications for the original extension, since it also performs manual gunzipping. will investigate